src/main/java/org/mozilla/taskcluster/client/temporarycredentials/Certificate.java

```diff
@@ -0,0 +1,130 @@
+package org.mozilla.taskcluster.client;
+
+import java.nio.ByteBuffer;
+
+import java.util.Date;
+import java.util.UUID;
+import java.security.NoSuchAlgorithmException;
+import java.security.InvalidKeyException;
+
+import javax.crypto.Mac;
+import javax.crypto.spec.SecretKeySpec;
+import java.io.UnsupportedEncodingException;
+
+import net.iharder.Base64;
+
+import com.google.gson.Gson;
+
+/*
+  This class handles creation of cretificates and generation of signatures.
+  Output is provided as JSON string.
+
+  constructors:
+    Default
+    Certificate(int version, Date start, Date expiry, String scopes[])
+    Certificate(int version, Date start, Date expiry, String scopes[], String clientId, String issuer)
+
+  public methods:
+  generateSignature(String)
+    This generates the signature based on the access token
+*/
+
+public class Certificate {
+
+  public int version = 1;
+  public String[] scopes;
+  public long start;
+  public long expiry;
+  public String seed;
+  public String signature = null;
+  private final transient String algo = "HmacSHA256";
+  //For named credentials
+  public String issuer = null;
+  public String clientId = null;
+
+  //constructors
+  public Certificate(){
+    signature = null;
+    generateSeed();
+  }
```

Comment:
Do we have a use case for the default constructor? I'm wondering if we could remove it, so that certificates can only be created if they provide all of the required fields.